### DISCUSSION TITLE
Types are not mentionned in Upgrading from v7...

### DISCUSSION THREAD

**Discussion by T0din** [gpbl/react-day-picker#1428] [Support] [answered]
and i believe, they should ^^

I'm trying to upgrade from v7 to v8 and types are an issue for Typescrit users (at least for me).

Could you update the documentation to help devs upgrade (if i'm not the only one) or just me if i'm doing it wrong please?

PS: i'm not used to asking for help on Github. If i forgot something or did it wrong, please tell me :) 

Thank you

`
ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/DatePicker.tsx
 ./src/components/DatePicker/DatePicker.tsx
 [tsl] ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/DatePicker.tsx(62,7)
       TS2322: Type 'Partial<Modifiers> | undefined' is not assignable to type 'Record<string, boolean | DateRange | Date | ((date: Date) => boolean) | Date[] | DateBefore | DateAfter | DateInterval | DayOfWeek | Matcher[]> | undefined'.
   Type 'Partial<Modifiers>' is not assignable to type 'Record<string, boolean | DateRange | Date | ((date: Date) => boolean) | Date[] | DateBefore | DateAfter | DateInterval | DayOfWeek | Matcher[]>'.
     Index signatures are incompatible.
       Type 'Matcher[] | undefined' is not assignable to type 'boolean | DateRange | Date | ((date: Date) => boolean) | Date[] | DateBefore | DateAfter | DateInterval | DayOfWeek | Matcher[]'.
         Type 'undefined' is not assignable to type 'boolean | DateRange | Date | ((date: Date) => boolean) | Date[] | DateBefore | DateAfter | DateInterval | DayOfWeek | Matcher[]'.
  @ ./src/components/DatePicker/DatePickerProvider.tsx 7:0-42 38:22-32
  @ ./src/components/DatePicker/RangeDatePicker/RangeDatePicker.tsx 20:0-59 57:15-33
  @ ./src/components/DatePicker/RangeDatePicker/index.ts 1:0-34 1:0-34
  @ ./src/components/DatePicker/index.ts 3:0-52 3:0-52
  @ ./src/components/index.ts 59:0-29 59:0-29
  @ ./src/index.ts 1:0-29 1:0-29
 ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/DatePicker.tsx
 ./src/components/DatePicker/DatePicker.tsx
 [tsl] ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/DatePicker.tsx(64,7)
       TS2322: Type 'string | string[] | undefined' is not assignable to type 'boolean | DateRange | Date | ((date: Date) => boolean) | Date[] | DateBefore | DateAfter | DateInterval | DayOfWeek | Matcher[] | undefined'.
   Type 'string' is not assignable to type 'boolean | DateRange | Date | ((date: Date) => boolean) | Date[] | DateBefore | DateAfter | DateInterval | DayOfWeek | Matcher[] | undefined'.
  @ ./src/components/DatePicker/DatePickerProvider.tsx 7:0-42 38:22-32
  @ ./src/components/DatePicker/RangeDatePicker/RangeDatePicker.tsx 20:0-59 57:15-33
  @ ./src/components/DatePicker/RangeDatePicker/index.ts 1:0-34 1:0-34
  @ ./src/components/DatePicker/index.ts 3:0-52 3:0-52
  @ ./src/components/index.ts 59:0-29 59:0-29
  @ ./src/index.ts 1:0-29 1:0-29
 ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/DatePicker.tsx
 ./src/components/DatePicker/DatePicker.tsx
 [tsl] ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/DatePicker.tsx(75,9)
       TS2322: Type '{ caption: string; month: string; months: string; day: string; weekday: string; disabled: string; outside: string; week: string; container: string; wrapper: string; interactionDisabled: string; navBar: string; ... 9 more ...; selected: string; }' is not assignable to type 'Partial<StyledElement<string>>'.
   Object literal may only specify known properties, and 'weekday' does not exist in type 'Partial<StyledElement<string>>'.
  @ ./src/components/DatePicker/DatePickerProvider.tsx 7:0-42 38:22-32
  @ ./src/components/DatePicker/RangeDatePicker/RangeDatePicker.tsx 20:0-59 57:15-33
  @ ./src/components/DatePicker/RangeDatePicker/index.ts 1:0-34 1:0-34
  @ ./src/components/DatePicker/index.ts 3:0-52 3:0-52
  @ ./src/components/index.ts 59:0-29 59:0-29
  @ ./src/index.ts 1:0-29 1:0-29
 ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/helpers.ts
 ./src/components/DatePicker/helpers.ts
 [tsl] ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/helpers.ts(15,3)
       TS2322: Type '{ [x: string]: DateRange | Date | undefined; }' is not assignable to type 'Partial<Modifiers>'.
   Index signatures are incompatible.
     Type 'DateRange | Date | undefined' is not assignable to type 'Matcher[] | undefined'.
       Type 'DateRange' is not assignable to type 'Matcher[]'.
  @ ./src/components/DatePicker/RangeDatePicker/RangeDatePicker.tsx 21:0-42 39:21-33
  @ ./src/components/DatePicker/RangeDatePicker/index.ts 1:0-34 1:0-34
  @ ./src/components/DatePicker/index.ts 3:0-52 3:0-52
  @ ./src/components/index.ts 59:0-29 59:0-29
  @ ./src/index.ts 1:0-29 1:0-29
 ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/DatePickerWithInput/DatePickerWithInput.tsx
 ./src/components/DatePicker/DatePickerWithInput/DatePickerWithInput.tsx
 [tsl] ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/DatePickerWithInput/DatePickerWithInput.tsx(40,16)
       TS2322: Type 'null' is not assignable to type 'Date | undefined'.
  @ ./src/components/DatePicker/DatePickerWithInput/index.ts 1:0-38 1:0-38
  @ ./src/components/DatePicker/index.ts 2:0-60 2:0-60
  @ ./src/components/index.ts 59:0-29 59:0-29
  @ ./src/index.ts 1:0-29 1:0-29
 ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/DatePickerWithInput/DatePickerWithInput.tsx
 ./src/components/DatePicker/DatePickerWithInput/DatePickerWithInput.tsx
 [tsl] ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/DatePickerWithInput/DatePickerWithInput.tsx(40,28)
       TS2322: Type 'null' is not assignable to type 'Date | undefined'.
  @ ./src/components/DatePicker/DatePickerWithInput/index.ts 1:0-38 1:0-38
  @ ./src/components/DatePicker/index.ts 2:0-60 2:0-60
  @ ./src/components/index.ts 59:0-29 59:0-29
  @ ./src/index.ts 1:0-29 1:0-29
 ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/IncludeAndExcludeDatePicker/IncludeAndExcludeDatePicker.tsx
 ./src/components/DatePicker/IncludeAndExcludeDatePicker/IncludeAndExcludeDatePicker.tsx
 [tsl] ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/IncludeAndExcludeDatePicker/IncludeAndExcludeDatePicker.tsx(2,10)
       TS2305: Module '"../../../../../../node_modules/react-day-picker/dist"' has no exported member 'DateUtils'.
  @ ./src/components/DatePicker/IncludeAndExcludeDatePicker/index.ts 1:0-46 1:0-46
  @ ./src/components/DatePicker/index.ts 5:0-46 5:0-46
  @ ./src/components/index.ts 59:0-29 59:0-29
  @ ./src/index.ts 1:0-29 1:0-29
 ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/RangeDatePicker/RangeDatePicker.tsx
 ./src/components/DatePicker/RangeDatePicker/RangeDatePicker.tsx
 [tsl] ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/RangeDatePicker/RangeDatePicker.tsx(2,21)
       TS2305: Module '"../../../../../../node_modules/react-day-picker/dist"' has no exported member 'DateUtils'.
  @ ./src/components/DatePicker/RangeDatePicker/index.ts 1:0-34 1:0-34
  @ ./src/components/DatePicker/index.ts 3:0-52 3:0-52
  @ ./src/components/index.ts 59:0-29 59:0-29
  @ ./src/index.ts 1:0-29 1:0-29
 ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/RangeDatePicker/helpers.ts
 ./src/components/DatePicker/RangeDatePicker/helpers.ts
 [tsl] ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/RangeDatePicker/helpers.ts(5,77)
       TS2322: Type '{ [x: string]: DateRange | Date | undefined; }' is not assignable to type 'Partial<Modifiers>'.
   Index signatures are incompatible.
     Type 'DateRange | Date | undefined' is not assignable to type 'Matcher[] | undefined'.
       Type 'DateRange' is missing the following properties from type 'Matcher[]': length, pop, push, concat, and 26 more.
  @ ./src/components/DatePicker/RangeDatePicker/index.ts 2:0-26 2:0-26
  @ ./src/components/DatePicker/index.ts 3:0-52 3:0-52
  @ ./src/components/index.ts 59:0-29 59:0-29
  @ ./src/index.ts 1:0-29 1:0-29
 ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/SingleDatePicker/SingleDatePicker.tsx
 ./src/components/DatePicker/SingleDatePicker/SingleDatePicker.tsx
 [tsl] ERROR in /Users/user/app/modules/common-ui/src/components/DatePicker/SingleDatePicker/SingleDatePicker.tsx(25,47)
       TS2322: Type '{ [x: string]: Date | undefined; }' is not assignable to type 'Partial<Modifiers>'.
   Index signatures are incompatible.
     Type 'Date | undefined' is not assignable to type 'Matcher[] | undefined'.
       Type 'Date' is missing the following properties from type 'Matcher[]': length, pop, push, concat, and 26 more.
  @ ./src/components/DatePicker/SingleDatePicker/index.ts 1:0-35 1:0-35
`

**Comment by gpbl** [comment, marked as answer]
@T0din have you removed the type package as described in https://react-day-picker.js.org/guides/upgrading ?

```
npm uninstall @types/react-day-picker
```

**Reply by T0din** [reply]
Hey @gpbl 

I did remove as adviced the @types

**Comment by T0din** [comment]
Hey @gpbl 

I don't know how status work in Github but my problem is definitely not solved.
I would hate for people to have the same issue and come here for answers and not find any

**Reply by gpbl** [reply]
@T0din please create a reproducible example of this problem: https://codesandbox.io/s/react-day-picker-8-vl4ry so that we can work it out. Thanks!
